iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/SubscriptionProvider.java

```diff
@@ -81,11 +87,49 @@ public synchronized void close() throws IoTDBConnectionException {
       throw new IoTDBConnectionException(e);
     } finally {
       super.close();
+      setUnavailable();
       isClosed.set(true);
     }
   }
 
   SubscriptionSessionConnection getSessionConnection() {
     return (SubscriptionSessionConnection) defaultSessionConnection;
   }
+
+  boolean isAvailable() {
+    return isAvailable.get();
+  }
+
+  void setAvailable() {
+    isAvailable.set(true);
+  }
+
+  void setUnavailable() {
+    isAvailable.set(false);
+  }
+
+  int getDataNodeId() {
+    return dataNodeId;
+  }
+
+  TEndPoint getEndPoint() {
+    return endPoint;
+  }
+
+  @Override
+  public String toString() {
+    return "{endPoint="
+        + endPoint
+        + ", dataNodeId="
+        + dataNodeId
+        + ", consumerId="
+        + consumerId
+        + ", consumerGroupId="
+        + consumerGroupId
+        + ", isAvailable="
+        + isAvailable
```

Comment:
No need since atomic variable class overrides `toString` the method...

```java
public String toString() {
	return Boolean.toString(this.get());
}
```

Reply:
Okey..